buildSrc/subprojects/kotlin-dsl/src/main/kotlin/plugins/kotlin-dsl-module.gradle.kts

```diff
@@ -30,7 +30,8 @@ plugins {
 
 // including all sources
 tasks.jar {
-    from(sourceSets.main.get().allSource)
+    val main by sourceSets.getting
+    from(main.allSource - main.resources)
```

Comment:
Is this the desired fix, or are we treating the symptoms here?
Would it be possible to do a general deduplication before packaging, instead of (or in addition to) this?
Would it make sense to add an integration test that tries to extract every generated artifact with a patched unzip?

Reply:
This is simply fixing a silly mistake in how the Kotlin DSL jars are assembled.

I'm not sure we want to do a general deduplication. We might want to enhance the archive tasks to allow failing on duplicate entries instead.

A distro integ test that validates that no shipped jar has duplicated entries could make sense too.

Reply:
Generalised deduplication (see #10005) would protect other projects. I found affected jars in ASM and Spring. Duplication seems to cause "grossly invalid zip files" to be written. Avoiding this should be the default. I do not know any project that does this on purpose.

Reply:
@bencaradocdavies, yes, see https://github.com/gradle/gradle/issues/10039